### DISCUSSION TITLE
MHIG_EU_AU_K1555_MU0415

### DISCUSSION THREAD

**Discussion by papirek99** [Mr-MIBonk/M.I.B._More-Incredible-Bash#468] [General]
Hello, someone here please patch  MHIG_EU_AU_K1555_MU0415  thanks 
![fe91b09b-d302-47c5-ba25-07c0e970e7f1](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/assets/26206774/03602836-4b92-4812-9ab8-b1d203dec57b)

[MU0415-JTB-07431.03.1587210369.log](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/files/15420412/MU0415-JTB-07431.03.1587210369.log)

**Comment by tl-tasmota** [comment]
GEM 3.1 is an old Version. Try the recent BashCode 3.7.1 which supports MHIG devices. No patched files needed. Patching works in the fly.

**Reply by papirek99** [reply]
is there any way to update gem 3.3 to 4.1??

**Comment by papirek99** [comment]
Thanks for the help BashCode 3.7.1 finally helped 
![2d61f961-d500-4aed-98c0-a47068fcfff3](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/assets/26206774/c7b04ae2-67e4-48ff-8b39-bc7d95d8e3fa)
![42be618a-1e3d-417a-83d8-0c881baa8680](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/assets/26206774/a053e269-dce3-4e10-a5bf-5ea752ada866)

**Comment by MajorSarcasm** [comment]
Hi. I have the same Train as this and have installed M.I.B. 3.7.1. After installation, rebooted and I still don't have 021000ee as an activation key. Not listed at all. Any ideas why this hasn't patched please?

**Reply by papirek99** [reply]
https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/releases/tag/V3.7.1

**Reply by MajorSarcasm** [reply]
Thanks for the reply, but what's the purpose of the link? I have 3.7.1 installed.

**Reply by Misterious30** [reply]
@MajorSarcasm did you get this fixed? Thanks :)

**Reply by MajorSarcasm** [reply]
@Misterious30 No, unfortunately not. I installed 3.7.1 but it did not auto patch.

**Reply by Misterious30** [reply]
Hey @MajorSarcasm thanks for coming back to me, I am struggling with this part as well its very frustrating.  @papirek99 did your M.I.B screen look like this? 

![PXL_20240920_210602926 MP](https://github.com/user-attachments/assets/b7ecc36e-3064-42d4-8b42-e2e2b95bebb3)

Whenever I press this PATCH NOW, it takes me to this screen and nothing happens:
![PXL_20240920_210435450 MP](https://github.com/user-attachments/assets/8ce015b9-d9cb-447f-915b-730f3d8439b7)

any ideas?

Many thanks for the your help in advance :)

**Comment by Misterious30** [comment]
Hi @papirek99 what exactly did you do inside the menu, once you loaded the latest GEM? I have access to GEM 4.12 and in the menu patch-aio --> PATCH NOW it does nothing. 
 thanks :)

**Comment by Misterious30** [comment]
Hey @MajorSarcasm I have patched my head unit this morning and it is showing 021000ee in the fec codes now. Can you tell me what firmware you have? 

I maybe able to help you? 

My issue was I was running the backup first and didn't let it finish!  Only when it says you can go back now, its complete.  I then ran patch now and it executed/applied the patch. 

I now need to load the maps onto the cars HDD

**Reply by MajorSarcasm** [reply]
@Misterious30 I'm running MHIG_EU_AU_K1555.

Glad you've managed to sort it. I'm sure I let the backup run to completion. I'll try it again.

**Comment by phantom83** [comment]
But why is it impossible to patch this firmware? I certainly have, and I've been trying for weeks, but to no avail. I've tried several versions of m.i.b., but it won't update the keys or apply the patch because it's not on the SD card, and I couldn't find it online. @Misterious30, which firmware did you fix? Can you send me a copy of the patch?

**Reply by Misterious30** [reply]
Hi @phantom83 I do not have the files anymore sorry! However I used Tag v3.7.1 of the Github repo code.  Are you trying to upgrade the firmware or add the fec code for lifetime maps?

**Reply by phantom83** [reply]
Thanks for your help. I managed to update the firmware. Now I'd like to add the FEC codes for the maps. I have OBDeleven if that can help.

**Reply by Misterious30** [reply]
Hi @phantom83 ODB will help with enabling the the GEM (Green engineering menu).  Once you have this, you have to access the GEM via the REM.  The script in the repo only works in the GEM but you must create a backup and wait till it says its complete before running the fec code patch. I believe there is a YOUTUBE video that Mr-Fix goes through on how to add the code. https://www.youtube.com/watch?v=rzOmVFBbDNg&t=0s - im assuming you have seen this already?
